### PR TITLE
Processing of box tests

### PR DESCRIPTION
**Pull request by anastasiiaSpaseeva**
Box and diagnostics tests are displayed;
Diagnostics tests are divided  by test cases;
Parsing of tests' helpers and unexpected behaviour is fixed;
Notification if sentence/paragraph is not found;

### CODE REVIEW

File: front-end/src/main/org/jetbrains/kotlin/spec/links/SpecPlaceHighlighter.kt [hunk 1]
```diff
@@ -14,14 +16,21 @@ data class SpecPlaceComponents(
         val sentenceNumber: Int? = null
 )
 
+
 object SpecPlaceHighlighter {
+    private const val paragraphNotFound = "Paragraph \"{1}\" of section \"{2}\" is not found."
+    private const val sentenceNotFound = "Sentence \"{1}\" is not found."
```
[the remark this answers is not in the document]
Comment (anastasiiaSpaseeva) [reply front-end/src/main/org/jetbrains/kotlin/spec/links/SpecPlaceHighlighter.kt]: done

File: front-end/src/main/org/jetbrains/kotlin/spec/links/SpecPlaceHighlighter.kt [hunk 2]
```diff
@@ -14,14 +16,21 @@ data class SpecPlaceComponents(
...
+
     private fun findParagraph(sectionId: String, paragraphNumber: Int): JQuery? {
-        val sectionElement = `$`("#$sectionId")
+        val sectionElement = `$`("#${sectionId.replace(".", """\.""")}")
```
[the remark this answers is not in the document]
Comment (anastasiiaSpaseeva) [reply front-end/src/main/org/jetbrains/kotlin/spec/links/SpecPlaceHighlighter.kt]: done

File: front-end/src/main/org/jetbrains/kotlin/spec/tests/SpecTestsLoader.kt [hunk 3]
```diff
@@ -120,22 +122,31 @@ class SpecTestsLoader(loadType: GithubTestsLoaderType) {
                 paragraphsInfo: List<Map<String, Any>>
         ) {
             val pathPrefix = "${sectionsPath.joinToString(".")}.$currentSection"
-            val tests = mutableMapOf<String, Any>()
 
+
+            val testsScope = mutableMapOf(
+                    TestTypeInfo.DIAG to getTestsByTestTypeInfo(responses, TestTypeInfo.DIAG),
+                    TestTypeInfo.BOX to getTestsByTestTypeInfo(responses, TestTypeInfo.BOX))
```
[the remark this answers is not in the document]
Comment (anastasiiaSpaseeva) [reply front-end/src/main/org/jetbrains/kotlin/spec/tests/SpecTestsLoader.kt]: done

File: front-end/src/main/org/jetbrains/kotlin/spec/tests/SpecTestsLoader.kt [hunk 4]
```diff
@@ -120,22 +122,31 @@ class SpecTestsLoader(loadType: GithubTestsLoaderType) {
...
+        }
+
+        /**
+         * @param responses
+         * @param testTypeInfo diagnostics or box type
+         *
+         * @return Map of tests depends on the testType
+         */
+        private fun getTestsByTestTypeInfo(responses: Array<out Map<String, Any>>, testTypeInfo: TestTypeInfo): MutableMap<String, Any> {
+            val testsDiag = mutableMapOf<String, Any>()
```
[the remark this answers is not in the document]
Comment (anastasiiaSpaseeva) [reply front-end/src/main/org/jetbrains/kotlin/spec/tests/SpecTestsLoader.kt]: done

File: front-end/src/main/org/jetbrains/kotlin/spec/tests/loaders/GithubTestsLoader.kt [hunk 5]
```diff
@@ -24,37 +26,60 @@ interface GithubTestsLoader {
...
+                override var type: String?
+                    get() = "GET"
+                    set(_) {}
+                override val error: ((jqXHR: JQueryXHR, textStatus: String, errorThrown: String) -> Any)?
+                    get() = { _, _, _ -> requestReject(Exception()) }
             }
         }
     }
 
+
```
[the remark this answers is not in the document]
Comment (anastasiiaSpaseeva) [reply front-end/src/main/org/jetbrains/kotlin/spec/tests/loaders/GithubTestsLoader.kt]: removed

File: front-end/src/main/org/jetbrains/kotlin/spec/tests/loaders/LoaderByTestsMapFile.kt [hunk 6]
```diff
@@ -3,13 +3,13 @@ package org.jetbrains.kotlin.spec.tests.loaders
...
 import kotlinx.serialization.json.JsonElement
-import kotlinx.serialization.json.JsonObject
 import org.jetbrains.kotlin.spec.tests.loaders.GithubTestsLoader.Companion.loadFileFromRawGithub
+import org.jetbrains.kotlin.spec.utils.TestTypeInfo
 import org.jetbrains.kotlin.spec.utils.format
 import kotlin.js.Promise
 
 
-class LoaderByTestsMapFile: GithubTestsLoader {
+class LoaderByTestsMapFile : GithubTestsLoader {
```
[the remark this answers is not in the document]
Comment (anastasiiaSpaseeva) [reply front-end/src/main/org/jetbrains/kotlin/spec/tests/loaders/LoaderByTestsMapFile.kt]: this whitespase is done by formatter

File: front-end/src/main/org/jetbrains/kotlin/spec/utils/helpers.kt [hunk 7]
```diff
@@ -36,7 +38,7 @@ fun <T> getValueByObjectPath(obj: Map<String, Any>, path: List<String>): T? {
 
     path.forEach {
         if (it !in soughtForObj) return null
-        soughtForObj = soughtForObj[it].unsafeCast<MutableMap<String, Any>>()
+        soughtForObj = soughtForObj[it].unsafeCast<MutableMap<String, Any>>() //todo  (<MutableMap<String, List<Any>>>)
```
[the remark this answers is not in the document]
Comment (anastasiiaSpaseeva) [reply front-end/src/main/org/jetbrains/kotlin/spec/utils/helpers.kt]: redundant todo is removed

File: front-end/src/main/org/jetbrains/kotlin/spec/utils/helpers.kt [hunk 8]
```diff
@@ -55,3 +57,11 @@ val Location.searchMap: MutableMap<String, String>
 
         return objURL
     }
+
+enum class TestTypeInfo(val path: String, val content: String, val contentPath: String) {
```
[the remark this answers is not in the document]
Comment (anastasiiaSpaseeva) [reply front-end/src/main/org/jetbrains/kotlin/spec/utils/helpers.kt]: ok, moved to utils.TestArea.kt

File: front-end/src/main/org/jetbrains/kotlin/spec/utils/helpers.kt [hunk 9]
```diff
@@ -1,5 +1,6 @@
 package org.jetbrains.kotlin.spec.utils
 
+import org.w3c.dom.CanvasDrawPath
```
[the remark this answers is not in the document]
Comment (anastasiiaSpaseeva) [reply front-end/src/main/org/jetbrains/kotlin/spec/utils/helpers.kt]: removed

File: front-end/src/main/org/jetbrains/kotlin/spec/links/SpecPlaceHighlighter.kt [hunk 10]
```diff
@@ -14,14 +15,22 @@ data class SpecPlaceComponents(
         val sentenceNumber: Int? = null
 )
 
+
 object SpecPlaceHighlighter {
+    private const val PARAGRAPH_NOT_FOUND = "Paragraph \"{1}\" of section \"{2}\" is not found."
+    private const val SENTENCE_NOT_FOUND = "Sentence \"{1}\" is not found."
+    private const val ESCAPED_SECTION_ID = """\."""
```
Comment (PetukhovVictor) [inline front-end/src/main/org/jetbrains/kotlin/spec/links/SpecPlaceHighlighter.kt]: It's just escaped separator. Maybe you mean `ESCAPED_SECTION_ID_SEPARATOR`?

**Pull request closed without merging**